Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -866,24 +876,27 @@ TEST_F(SerializerTest, FailOnInvalidInputBytes) {
 }
 
 TEST_F(SerializerTest, EncodesKey) {
-  EXPECT_EQ("projects/p/databases/d/documents", serializer.EncodeKey(Key("")));
+  EXPECT_EQ("projects/p/databases/d/documents",
+            FromBytes(serializer.EncodeKey(Key(""))));
   EXPECT_EQ("projects/p/databases/d/documents/one/two/three/four",
-            serializer.EncodeKey(Key("one/two/three/four")));
+            FromBytes(serializer.EncodeKey(Key("one/two/three/four"))));
```

Comment:
This arrangement leaks the `pb_bytes_array_t` allocated by `EncodeKey`.

Could you work around this adding a variant of `FromBytes` that takes `pb_bytes_array_t*&&`?

Alternatively, we could add a member `EncodeKey` to the fixture that both runs through the serializer's `EncodeKey`, the converts to string, then frees the intermediate value.

Reply:
Thanks for noticing!

Actually, I don't need a non-ownership-taking overload of `FromBytes` at the moment. I'm using `ByteString` to deal with memory management here.

Reply:
There's a good case to be made for just having the Encoder produce ByteStrings in the first place, and then using release() to give away the pointer when assigning it into a structure. This would make the arrangement safe by default which is likely to be better for us in the long run.

WDYT?

Reply:
I fully agree, though I'd prefer to leave it off to a future PR.